framework/src/org/checkerframework/framework/test/TestConfigurationBuilder.java

```diff
@@ -0,0 +1,363 @@
+package org.checkerframework.framework.test;
+
+import org.checkerframework.framework.util.PluginUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class TestConfigurationBuilder {
+
+
+    private List<File> diagnosticFiles;
+    private List<File> testSourceFiles;
+    private Set<String> processors;
+    private SimpleOptionMap options;
+
+    private boolean shouldEmitDebugInfo;
+
+    public TestConfigurationBuilder() {
+        diagnosticFiles = new ArrayList<>();
+        testSourceFiles = new ArrayList<>();
+        processors = new LinkedHashSet<>();
+        options = new SimpleOptionMap();
+        shouldEmitDebugInfo = false;
+    }
+
+    public <T extends Collection<E>, E>  T addAll(T coll, Iterable<? extends E> objs) {
```

Comment:
Should this method be static? private?
The method modifies the first parameter. None of the invocations of the method that I see in this diff use the returned value. It would be clearer to return void and document the modification of the parameter.